components/director/internal/domain/formation/service.go

```diff
@@ -610,3 +652,25 @@ func (s *service) getAvailableScenarios(ctx context.Context, tenantID string) ([
 	}
 	return out, nil
 }
+
+func (s *service) createFormation(ctx context.Context, tenant, templateName, formationName string) error {
+	fTmpl, err := s.formationTemplateRepository.GetByName(ctx, templateName)
+	if err != nil {
+		log.C(ctx).Errorf("An error occurred while getting formation template by name: %q: %v", templateName, err)
+		return errors.Wrapf(err, "An error occurred while getting formation template by name: %q", templateName)
+	}
+
+	formation := &model.Formation{
+		ID:                  s.uuidService.Generate(),
+		TenantID:            tenant,
+		FormationTemplateID: fTmpl.ID,
+		Name:                formationName,
+	}
+	log.C(ctx).Debugf("Creating formation with name: %q and template ID: %q...", formationName, fTmpl.ID)
+	if err = s.formationRepository.Create(ctx, formation); err != nil {
+		log.C(ctx).Errorf("An error occurred while creating formation with name: %q and template ID: %q", formationName, fTmpl.ID)
+		return errors.Wrapf(err, "An error occurred while reating formation with name: %q and template ID: %q", formationName, fTmpl.ID)
+	}
+
+	return nil
```

Comment:
If the previous comment is applicable, it could return `(*Formation, error)`

Reply:
answered above

Reply:
answered above